### PR TITLE
Inline editing table update

### PR DESCRIPTION
**Pull request by iaincrawf**
Add and delete buttons + tests and stories, shift + tab tabs back the way through the table

### CODE REVIEW

File: src/stories/TableWithInlineEditing.stories.js [hunk 1]
```diff
@@ -70,19 +69,33 @@ storiesOf('TableWithInlineEditing', module)
             <div>{story()}</div>
         </ThemeProvider>
     ))
-    .add('allowed to edit', () => (
+    .add('allowed to edit, add and delete', () => (
+        <TableWithInlineEditing
+            columnsInfo={columnsInfo}
+            content={content}
+            updateContent={{}}
+            allowedToEdit
```
Comment (lewisrenfrew) [inline src/stories/TableWithInlineEditing.stories.js]: updateContent should be a function.  You can just do 

`updateContent={() => {}} `

or similiar

File: src/stories/TableWithInlineEditing.stories.js [hunk 2]
```diff
@@ -70,19 +69,33 @@ storiesOf('TableWithInlineEditing', module)
...
+            allowedToCreate
+            allowedToDelete
+        />
+    ))
+    .add('allowed to edit and add', () => (
         <TableWithInlineEditing
             columnsInfo={columnsInfo}
             content={content}
             updateContent={{}}
             allowedToEdit
```
Comment (lewisrenfrew) [inline src/stories/TableWithInlineEditing.stories.js]: same again

File: src/stories/TableWithInlineEditing.stories.js [hunk 3]
```diff
@@ -70,19 +69,33 @@ storiesOf('TableWithInlineEditing', module)
...
-    .add('without edit permission', () => (
+    .add('allowed to edit only', () => (
         <TableWithInlineEditing
             columnsInfo={columnsInfo}
             content={content}
             updateContent={{}}
-            allowedToEdit={false}
+            allowedToEdit
         />
+    ))
```
Comment (lewisrenfrew) [inline src/stories/TableWithInlineEditing.stories.js]: same again

File: src/stories/TableWithInlineEditing.stories.js [hunk 4]
```diff
@@ -70,19 +69,33 @@ storiesOf('TableWithInlineEditing', module)
...
+    .add('Not allowed to edit', () => (
+        <TableWithInlineEditing columnsInfo={columnsInfo} content={content} updateContent={{}} />
```
Comment (lewisrenfrew) [inline src/stories/TableWithInlineEditing.stories.js]: same again. Also you should pass 

`allowedToEdit={false}`

or perhaps make false the defaultProp to reduce keystrokes?

File: src/stories/TableWithInlineEditing.stories.js (hunk 1 above)
Comment (iaincrawf) [reply src/stories/TableWithInlineEditing.stories.js]: Ahh yeah. Now yours is through I'll merge the fixes in to this stuff, will make sure this stuff is in the merged version

File: src/components/TableWithInlineEditing.js [hunk 5]
```diff
@@ -130,67 +168,75 @@ const Row = ({
...
-            </TableRow>
-        </Fragment>
+                                    ) : (
+                                        <div id={`inner${rowIndex}-${index}`}>
+                                            {column.type === 'dropdown' ? (
+                                                <Dropdown
+                                                    onChange={handleCellChange}
+                                                    items={column.options}
+                                                    value={rowContent[column.key]}
+                                                    propertyName={column.key}
```
Comment (lewisrenfrew) [inline src/components/TableWithInlineEditing.js]: Not really a change request just an observation.... Dropdown takes a new optional 'allowNoValue' prop here which defaults to true. In its current form, every dropdown the table ever renders will allow a blank value. This seems reasonable to me but not sure if your use case requires restricting of blank values? If it does you might need to handle this at your component level.

Comment (iaincrawf) [reply src/components/TableWithInlineEditing.js]: hmm, I think I might pass in false from this component for now as I don't think I'd want blank values in my own use case, then if we need the option in future we can change the table component to take in whether it allows blank for dropdowns

**Pull request merged**